misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -114,11 +148,50 @@
 
         LinePosition linePosition = LinePosition.from(context.cursorPosition().getLine(), context.cursorPosition().
                 getCharacter());
-        String symbolAtCursor = semanticModel.symbol(srcFile.get(), linePosition).get().name();
 
-        String generatedFunctionName =
-                String.format("map%sTo%s(%s)", foundTypeRight, foundTypeLeft, symbolAtCursor);
-        fEdits.add(new TextEdit(newTextRange, generatedFunctionName));
+        Symbol symbolAtCursor = semanticModel.symbol(srcFile.get(), linePosition).get();
+
+
+        String symbolAtCursorName = symbolAtCursor.name();
+        String symbolAtCursorType = SymbolUtil.getTypeDescriptor(symbolAtCursor).get().typeKind().toString();
+
+        String generatedFunctionName;
+        // Insert function call in the code where error is found
+        switch (symbolAtCursorType) {
+            case "RECORD":
```

Comment:
usually these type of labels we should be able to get from `symbolAtCursorType` class or some relevant class as constants.
eg.  you have used in line 136 `DiagnosticPropertyKind.SYMBOLIC`
we need to reuse them, so if they change those labels, our code will not break